packages/venia-ui/lib/components/Image/image.js

```diff
@@ -75,18 +113,48 @@ const Image = props => {
     );
 };
 
+const conditionallyRequiredString = (props, propName, componentName) => {
+    // This component needs one of src or resource to be provided.
+    if (!props.src && !props.resource) {
+        return new Error(
+            `Missing both 'src' and 'resource' props in ${componentName}. ${componentName} needs at least one of these to be provided.`
+        );
+    }
+
+    return PropTypes.checkPropTypes(
+        {
+            resource: string,
+            src: string
+        },
+        props,
+        propName,
+        componentName
+    );
+};
+
 Image.propTypes = {
     alt: string,
     classes: shape({
         loaded: string,
         notLoaded: string,
         root: string
     }),
+    height: oneOfType([number, string]),
     onError: func,
     onLoad: func,
     placeholder: string,
-    src: string,
-    fileSrc: string
+    resource: conditionallyRequiredString,
+    resourceHeight: oneOfType([number, string]),
+    resourceWidth: oneOfType([number, string]),
+    sizes: string,
+    src: conditionallyRequiredString,
+    type: string,
+    width: oneOfType([number, string])
+};
+
+Image.defaultProps = {
+    placeholder: transparentPlaceholder,
```

Comment:
The problem is that this placeholder is very specific to venia sample data. IMO the components should provide this or it should just not have one.

Reply:
I'm glad you brought this up, I thought about that too.

1. Every component I ran into that was passing in `placeholder` was passing in the same one
2. Some of them weren't passing in any placeholders (we lost this functionality. Do we want it back?)

I do see the downside that non-Venia consumers of our `Image` component may want a different size placeholder though.

Reply:
Yea! As I said in the other comment (playing ping pong now!) we are just coincidentally only using `Image` for product images but it's supposed to be more generic.

Reply:
Okay, `Image` should be able to handle anything now. We still bundle & use the 4x5 transparent image but we can size it however we want now.

https://github.com/magento/pwa-studio/pull/1871/commits/50761a370eaf396a6eec1e489e3766b2f430e096